src/lib/seam/connect/seam-http-request.ts

```diff
@@ -46,22 +46,23 @@ export class SeamHttpRequest<
 
   public get url(): URL {
     const { client } = this.#parent
-    const baseUrl = client.defaults.baseURL
-    if (baseUrl === undefined) {
-      throw new Error('baseUrl is required')
-    }
-
-    const params = this.#config.params
-    if (params === undefined) {
-      return new URL(this.#config.path, baseUrl)
-    }
+    const { params } = this.#config
 
     const serializer =
       typeof client.defaults.paramsSerializer === 'function'
         ? client.defaults.paramsSerializer
         : serializeUrlSearchParams
 
-    return new URL(`${this.#config.path}?${serializer(params)}`, baseUrl)
+    const origin = getUrlPrefix(client.defaults.baseURL ?? '')
+
+    const pathname = this.#config.path.startsWith('/')
+      ? this.#config.path
+      : `/${this.#config.path}`
+
+    const path =
+      params == null ? pathname : [pathname, serializer(params)].join('?')
```

Comment:
why not a template literal?

```suggestion
      params == null ? pathname : `${pathname}?${serializer(params)}`
```

Reply:
I go back and forth on this, sometimes join is just easier to read, we can switch though.

One case where I always use join is with more than two values that all join with the same separator.